Marlin/src/Marlin.cpp

```diff
@@ -396,48 +422,51 @@ void disable_all_steppers() {
 #endif
 
 #if ENABLED(HOST_PROMPT_SUPPORT)
+
+  inline void say_m876_response(const char * const pstr) {
+    SERIAL_ECHOPGM("M876 Responding PROMPT_");
+    serialprintPGM(pstr);
+    SERIAL_EOL();
+  }
+
   char host_prompt_reason = PROMPT_NOT_DEFINED;
-  void host_response_handler(char c) { host_response_handler((int)strtol(&c, NULL, 1)); }
-  void host_response_handler(int response) {
-    switch(host_prompt_reason)
-    {
+  void host_response_handler(const uint8_t response) {
+    switch (host_prompt_reason) {
       case PROMPT_FILAMENT_RUNOUT_TRIPPED:
-        if(response==0) {
+        if (response == 0)
           advanced_pause_menu_response = ADVANCED_PAUSE_RESPONSE_EXTRUDE_MORE;
```

Comment:
HOST_PROMPT_SUPPORT can be active, but it is not said to have ADVANCED_PAUSE_FEATURE enabled.
You could have FILAMENT_RUNOUT_SENSOR active to have everything managed by the host and not have ADVANCED_PAUSE_FEATURE, this means a series of checks that are missing, but honestly it seems to me all a bit too complex.

Reply:
This part will likely change, but at the moment this is set to this value only in the scenario where its being used and all those conditions exist, Id like to break the reliance on the LCD stuff, but lets get it working and tested first!

Reply:
Honestly, I thought of it in a different way. I print from HOST (server / octoprint), the firmware detects the filament end and tells it to the host that it should be him to manage everything. So once the command  // action: is sent, it is the host that takes care of making requests to the user, sending to the firmware the same requests to put on the display so that the user can also respond locally.
So instead the firmware is very complicated, because it may or may not have a function if it has that function there are instructions on the display and send the same instructions from the firmware to the host, is likely to make the firmware too large, in addition to the fact that there are a more and more of checks to check!!!
So either enable advanced pauses to manage everything from the display if I do not have a special host, or enable support for the host prompt, and the only thing the firmware needs to do is report the filament, then it is the host that it does everything.

Reply:
The point is for the firmware to tell the host how it wants to handle it. This way, it can still behave nicely if there is no host, but if the host is configured to do it, it can request control and handle it. Aim is to make it seamless. Yes, it's creeping up in size a bit but still not an issue on a 2560 even with tons of options enabled.